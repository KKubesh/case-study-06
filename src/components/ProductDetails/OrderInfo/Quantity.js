import React, {Component} from 'react';
import plus from '../../../images/plus.png';
import minus from '../../../images/minus.png';


class Quantity extends Component {
    render () {
        return (
            <div className="quantity" data-testid="quantity">
                quantity:
                <div className="quantity-buttons">
                    <img src={minus} alt="remove item"/>
                    1
                    <img src={plus} alt="add item"/>
                </div>
            </div>
                
        )
    }
}

export default Quantity;