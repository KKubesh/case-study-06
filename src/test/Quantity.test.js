import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Quantity from '../components/ProductDetails/OrderInfo/Quantity';

afterEach(cleanup);

test('<Quantity />', () => {
    const { getByTestId, container } = render(<Quantity/>);
    expect(getByTestId('quantity')).toBeTruthy();
    expect(getByTestId('quantity').textContent).toBe("quantity:1");

    expect(container.firstChild).toMatchSnapshot();
});