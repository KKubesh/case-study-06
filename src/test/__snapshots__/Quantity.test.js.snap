// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quantity /> 1`] = `
<div
  class="quantity"
  data-testid="quantity"
>
  quantity:
  <div
    class="quantity-buttons"
  >
    <img
      alt="remove item"
      src="minus.png"
    />
    1
    <img
      alt="add item"
      src="plus.png"
    />
  </div>
</div>
`;
